src/components/main-panel/main-panel.tsx

```diff
@@ -73,11 +84,30 @@ export const MainPanel = () => {
         })
     );
 
+    const handleModalOnClick = (element?: TElement) => {
+        const successToast = {
+            message: "Changes discarded.",
+            type: EToastTypes.SUCCESS_TOAST,
+        };
+        removeFocusedElement();
+        focusElement(element);
+        hideModal();
+        showToast(successToast);
+    };
+
     // =========================================================================
     // EVENT HANDLERS
     // =========================================================================
     const handleElementCardClick = (element: TElement) => () => {
-        focusElement(element);
+        const newModal = {
+            type: EModalType.DiscardChanges,
+            onClickActionButton: () => handleModalOnClick(element),
+        };
+        if (isDirty) {
+            showModal(newModal);
+        } else {
+            focusElement(element);
+        }
```

Comment:
this is causing the modal to show even if I am clicking the same card in the main panel

Reply:
if I have unsaved changes, on clicking the same element in the main panel, the save changes button got updated to `Saved`

can we also stop the `focusElement()` call in such a scenario?

Reply:
while this has prevented the clicking of the same element, this still allowed me to switch elements with unsaved changes

even though the discard changes modal appeared, the element is switched despite clicking "Keep editing" button

Reply:
updated in latest changes

Reply:
without `focusElement()`, the cards in the main panel cannot be focused anymore. this prevents admins from clicking a card when none are focused or switching between cards when there are no unsaved changes.

Reply:
seems like the latest fix brought back an older bug:
> if I have unsaved changes, on clicking the same element in the main panel, the save changes button got updated to Saved

behaviour should be:
- admin should be able to focus an element
- modal should show if field is dirty and admin closes the edit element panel
- modal should show if field is dirty and admin focuses on another panel
- modal should not show if admin clicks an already focused element
- clicking on an already focused dirtied element should retain the `Save Changes` button